packages/lesswrong/lib/collections/typingIndicators/collection.ts

```diff
@@ -0,0 +1,36 @@
+import {addUniversalFields, getDefaultMutations, getDefaultResolvers} from "../../collectionUtils";
+import {MutationOptions} from "../../vulcan-core/default_mutations";
+import {createCollection} from "../../vulcan-lib";
+import {userOwns} from "../../vulcan-users/permissions";
+import schema from "./schema";
+
+const options: MutationOptions<DbTypingIndicator> = {
+  newCheck: (user: DbUser|null, document: DbTypingIndicator|null) => {
+    if (!user || !document) return false;
+    return true
+  },
+
+  editCheck: (user: DbUser|null, document: DbTypingIndicator|null) => {
+    if (!user || !document) return false;
+    return userOwns(user, document) 
+  },
+
+  removeCheck: (user: DbUser|null, document: DbTypingIndicator|null) => {
+    if (!user || !document) return false;
+    return userOwns(user, document) 
+  },
+}
+
+export const TypingIndicators: TypingIndicatorsCollection = createCollection({
+  collectionName: 'TypingIndicators',
+  typeName: 'TypingIndicator',
+  collectionType: 'pg',
+  schema,
+  resolvers: getDefaultResolvers('TypingIndicators'),
+  mutations: getDefaultMutations('TypingIndicators', options),
+  logChanges: true,
+})
+
```

Comment:
I think you also want to define a `TypingIndicators.checkAccess` which default-denies access via our GraphQL API, since legit access will only be going through direct DB query inside the SSE logic